auth-api/src/main/java/io/stargate/auth/model/Credentials.java

```diff
@@ -43,10 +44,21 @@ public void setPassword(String password) {
     this.password = password;
   }
 
+  @JsonCreator
+  public Credentials(
+      @JsonProperty("username") String username, @JsonProperty("password") String password) {
+    this.username = username;
+    this.password = password;
+  }
+
   @Override
   public boolean equals(Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
     Credentials that = (Credentials) o;
     return Objects.equals(username, that.username) && Objects.equals(password, that.password);
```

Comment:
does Objects.equals do the same thing as `StringUtils.equals` (tests equality, null-safe)?

Reply:
Objects.equals was an IntelliJ recommendation for null safety.